test/random_dem.cpp

```diff
@@ -521,6 +576,8 @@ int32_t random_dem_test(ptrdiff_t dims[2], uint32_t seed) {
   delete[] marks;
   delete[] accum;
   delete[] target;
+  delete[] gradient;
+  // TODO: delete cellsize and unit?
```

Comment:
I'm unsure if the `float cellsize` and `char unit` should also be deleted here?

Reply:
You only need to `delete` things that you allocated with `new` (it's like malloc/free in C).